libcaf_net/caf/net/socket_manager.hpp

```diff
@@ -44,17 +44,17 @@ class CAF_NET_EXPORT socket_manager : public detail::atomic_ref_counted,
   /// Returns the handle for the managed socket.
   virtual socket handle() const = 0;
 
-  /// @private
-  virtual void handle(socket new_handle) = 0;
-
-  /// Returns a reference to the hosting @ref actor_system instance.
-  virtual actor_system& system() noexcept = 0;
-
   /// Returns the owning @ref multiplexer instance.
-  virtual multiplexer& mpx() noexcept = 0;
+  multiplexer& mpx() {
+    CAF_ASSERT(mpx_ptr());
+    return *mpx_ptr();
+  }
 
   /// Returns the owning @ref multiplexer instance.
-  virtual const multiplexer& mpx() const noexcept = 0;
+  const multiplexer& mpx() const {
```

Comment:
Actually, why not simplify this further? There's no reason for us to propagate `const` to the multiplexer in the first place:

```c++
  multiplexer& mpx() const;
  virtual multiplexer* mpx_ptr() const noexcept = 0;
```

Reply:
Wouldn't this break encapsulation, you could do a mutable operation on the multiplexer from a const declared context?

Reply:
The `multiplexer` is not part of the socket manager. The socket manager merely points to it. If you modify the `multiplexer`, it has no effect on the socket manager. No encapsulation broken, because the manager does not encapsulate a multiplexer. 🙂

Reply:
That's a valid viewpoint. I'll rebase and push.